osu.Game/Overlays/SkinEditor/SkinSelectionHandler.cs

```diff
@@ -31,8 +32,38 @@ public partial class SkinSelectionHandler : SelectionHandler<ISerialisableDrawab
             UpdatePosition = updateDrawablePosition
         };
 
+        private bool allSelectedSupportManualSizing => SelectedItems.All(b => (b as CompositeDrawable)?.AutoSizeAxes == Axes.None);
+
         public override bool HandleScale(Vector2 scale, Anchor anchor)
         {
+            bool adjustSize;
+
+            switch (anchor)
+            {
+                // for corners, adjust scale.
+                case Anchor.TopLeft:
+                case Anchor.TopRight:
+                case Anchor.BottomLeft:
+                case Anchor.BottomRight:
+                    adjustSize = false;
+                    break;
+
+                // for edges, adjust size.
+                case Anchor.TopCentre:
+                case Anchor.CentreLeft:
+                case Anchor.CentreRight:
+                case Anchor.BottomCentre:
+                    // autosize elements can't be easily handled so just disable sizing for now.
+                    if (!allSelectedSupportManualSizing)
+                        return false;
+
+                    adjustSize = true;
+                    break;
```

Comment:
This should be checking if all elements in there support sizing along the given axis, not just checking if they're completely autosized.

This being written the way it is, is probably the root cause for all of the following funky behaviours:

https://github.com/ppy/osu/assets/20418176/11026a4e-fa8d-4c14-b9dd-4d658f706051

Reply:
For reference, I mean something along the lines of

```diff
diff --git a/osu.Game/Overlays/SkinEditor/SkinSelectionHandler.cs b/osu.Game/Overlays/SkinEditor/SkinSelectionHandler.cs
index 77e522a925..a4d530a4d9 100644
--- a/osu.Game/Overlays/SkinEditor/SkinSelectionHandler.cs
+++ b/osu.Game/Overlays/SkinEditor/SkinSelectionHandler.cs
@@ -32,11 +32,11 @@ public partial class SkinSelectionHandler : SelectionHandler<ISerialisableDrawab
             UpdatePosition = updateDrawablePosition
         };
 
-        private bool allSelectedSupportManualSizing => SelectedItems.All(b => (b as CompositeDrawable)?.AutoSizeAxes == Axes.None);
+        private bool allSelectedSupportManualSizing(Axes axis) => SelectedItems.All(b => (b as CompositeDrawable)?.AutoSizeAxes.HasFlagFast(axis) == false);
 
         public override bool HandleScale(Vector2 scale, Anchor anchor)
         {
-            bool adjustSize;
+            Axes adjustAxis;
 
             switch (anchor)
             {
@@ -45,19 +45,25 @@ public override bool HandleScale(Vector2 scale, Anchor anchor)
                 case Anchor.TopRight:
                 case Anchor.BottomLeft:
                 case Anchor.BottomRight:
-                    adjustSize = false;
+                    adjustAxis = Axes.Both;
                     break;
 
                 // for edges, adjust size.
+                // autosize elements can't be easily handled so just disable sizing for now.
                 case Anchor.TopCentre:
+                case Anchor.BottomCentre:
+                    if (!allSelectedSupportManualSizing(Axes.Y))
+                        return false;
+
+                    adjustAxis = Axes.Y;
+                    break;
+
                 case Anchor.CentreLeft:
                 case Anchor.CentreRight:
-                case Anchor.BottomCentre:
-                    // autosize elements can't be easily handled so just disable sizing for now.
-                    if (!allSelectedSupportManualSizing)
+                    if (!allSelectedSupportManualSizing(Axes.X))
                         return false;
 
-                    adjustSize = true;
+                    adjustAxis = Axes.X;
                     break;
 
                 default:
@@ -151,10 +157,20 @@ public override bool HandleScale(Vector2 scale, Anchor anchor)
                 if (Precision.AlmostEquals(MathF.Abs(drawableItem.Rotation) % 180, 90))
                     currentScaledDelta = new Vector2(scaledDelta.Y, scaledDelta.X);
 
-                if (adjustSize)
-                    drawableItem.Size *= currentScaledDelta;
-                else
-                    drawableItem.Scale *= currentScaledDelta;
+                switch (adjustAxis)
+                {
+                    case Axes.X:
+                        drawableItem.Width *= currentScaledDelta.X;
+                        break;
+
+                    case Axes.Y:
+                        drawableItem.Height *= currentScaledDelta.Y;
+                        break;
+
+                    case Axes.Both:
+                        drawableItem.Scale *= currentScaledDelta;
+                        break;
+                }
             }
 
             return true;
@@ -203,7 +219,8 @@ protected override void OnSelectionChanged()
         {
             base.OnSelectionChanged();
 
-            SelectionBox.CanScaleX = SelectionBox.CanScaleY = allSelectedSupportManualSizing;
+            SelectionBox.CanScaleX = allSelectedSupportManualSizing(Axes.X);
+            SelectionBox.CanScaleY = allSelectedSupportManualSizing(Axes.Y);
             SelectionBox.CanScaleProportionally = true;
             SelectionBox.CanFlipX = true;
             SelectionBox.CanFlipY = true;

```

This fixes a crash, too; to see it, try resizing any health bar (the argon one will do) vertically or horizontally. It'll die due to trying to set `Size` on a half-autosized component which is not allowed.